### PR TITLE
Add `eslint` and `stylelint`

### PR DESCRIPTION
**Pull request by andyduong1920**
https://github.com/nimblehq/elixir-templates/issues/85
https://github.com/nimblehq/elixir-templates/issues/109

## What happened

- Add **eslint** and **stylelint**.
- Config the **stylelint** order follow **smacss**.
- Refactor some modules.

## Insight

1/ For eslint and stylelint configurations, mostly copy from the existing projects, feel free to suggest any other default configuration 🤗
2/ For the refactor part.
2.1/ Remove some unnecessary `\"` 
2.2/ Inject the command to the beginning of `codebase` or `codebase.fix` block, so for that, we could remove some dependency in the test (for example for the sobelow test, we do not need to depend on the prettier module) -> Solved [Ros's concern](https://nimble-co.slack.com/archives/C0112R61J3W/p1640169951004700?thread_ts=1640169235.003900&cid=C0112R61J3W) 

3/ Temporary skip the StyleLint for the default Phoenix files

image

We don't spend time to fix these files as we will generate another structure for CSS on another PR https://github.com/nimblehq/elixir-templates/issues/100 later

## Proof Of Work

The test is passed

### CODE REVIEW

File: lib/nimble_template/addons/variants/phoenix/web/es_lint.ex [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    Generator.replace_content(
+      "assets/js/app.js",
+      "window.addEventListener(\"phx:page-loading-start\", info => NProgress.start())",
+      "window.addEventListener(\"phx:page-loading-start\", _info => NProgress.start())"
+    )
+
+    Generator.replace_content(
+      "assets/js/app.js",
+      "window.addEventListener(\"phx:page-loading-stop\", info => NProgress.done())",
+      "window.addEventListener(\"phx:page-loading-stop\", _info => NProgress.done())"
```
Comment (rosle) [inline lib/nimble_template/addons/variants/phoenix/web/es_lint.ex:93]: Not sure if we need this. Are they going to be fixed after we fix codebase on `post_apply/1`?

File: priv/templates/nimble_template/assets/.stylelintrc.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "extends": ["stylelint-config-sass-guidelines", "stylelint-config-property-sort-order-smacss"],
```
Comment (rosle) [inline priv/templates/nimble_template/assets/.stylelintrc.json:2]: How about

```suggestion
  "extends": [
    "stylelint-config-sass-guidelines", 
    "stylelint-config-property-sort-order-smacss"
  ],
```

File: test/nimble_template/addons/test_env_test.exs [hunk 3]
```diff
@@ -20,7 +20,10 @@ defmodule NimbleTemplate.Addons.TestEnvTest do
           assert file =~ """
                    defp aliases do
                      [
-                       codebase: [\"deps.unlock --check-unused\", \"format --check-formatted\"],
+                       codebase: [
+                         \"deps.unlock --check-unused\",
+                         \"format --check-formatted\"
```
Comment (rosle) [inline test/nimble_template/addons/test_env_test.exs:25]: Does this work?
```suggestion
                         "deps.unlock --check-unused",
                         "format --check-formatted"
```

File: test/nimble_template/addons/credo_test.exs [hunk 4]
```diff
@@ -41,7 +41,11 @@ defmodule NimbleTemplate.Addons.CredoTest do
           assert file =~ """
                    defp aliases do
                      [
-                       codebase: [\"deps.unlock --check-unused\", \"format --check-formatted\", \"credo --strict\"],
+                       codebase: [
+                         "credo --strict",
+                         "deps.unlock --check-unused",
+                         "format --check-formatted"
```
Comment (rosle) [inline test/nimble_template/addons/credo_test.exs:47]: Do we need to remove this on this test based on our new update? 💡

File: lib/nimble_template/addons/variants/phoenix/web/es_lint.ex (hunk 1 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/web/es_lint.ex:93]: the eslint doesn't fix this @rosle :D

File: priv/templates/nimble_template/assets/.stylelintrc.json (hunk 2 above)
Comment (andyduong1920) [reply priv/templates/nimble_template/assets/.stylelintrc.json]: yeah, I have updated it in https://github.com/nimblehq/elixir-templates/pull/144/commits/40ded4d770d9f74aa6699fe458fed4922d7b8bac Ros 👍

File: test/nimble_template/addons/test_env_test.exs (hunk 3 above)
Comment (andyduong1920) [reply test/nimble_template/addons/test_env_test.exs]: Good catch @rosle I have updated it in https://github.com/nimblehq/elixir-templates/pull/144/commits/5df253ce7739fb4b5a5712e1e6f17d0700b20302

File: test/nimble_template/addons/credo_test.exs (hunk 4 above)
Comment (andyduong1920) [reply test/nimble_template/addons/credo_test.exs]: Yeah, we can remove these assert, I have removed it in https://github.com/nimblehq/elixir-templates/pull/144/commits/88c7a3ec017f96f8340b525456ab2a2fc55a85f8

File: priv/templates/nimble_template/assets/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "env": {
+    "browser": true,
+    "node": true,
+    "es2021": true
+  },
+  "extends": [
+    "eslint:recommended",
+    "plugin:prettier/recommended"
```
Comment (rosle) [inline priv/templates/nimble_template/assets/.eslintrc.json:9]: I'm not sure, with this change, do we still need the `prettier` addon? 🤔

**Review by bterone — APPROVED** [review]

**Review by rosle — APPROVED** [review]

File: priv/templates/nimble_template/assets/.eslintrc.json (hunk 5 above)
Comment (andyduong1920) [reply priv/templates/nimble_template/assets/.eslintrc.json:9]: Yes @rosle although this config will install the `prettier` (if it isn't installed yet) but our prettier module contains more topics than just `install`, we have
- prettier-plugin-eex
- add prettier, prettier.fix commands, alias

So we still need the `prettier` addon module there FYU

**Pull request merged**